torch/distributed/nn/_functions.py

```diff
@@ -0,0 +1,339 @@
+    in ``tensor_list`` should reside on a separate GPU
+
+    Only the GPU of ``tensor_list[dst_tensor]`` on the process with rank ``dst``
+    is going to receive the final result.
+
+    Only nccl backend is currently supported
+    tensors should only be GPU tensors
+
+    Arguments:
+        tensor_list (List[Tensor]): Input GPU tensors of the collective.
+            You also need to make sure that ``len(tensor_list)`` is the same for
+            all the distributed processes calling this function.
+        dst (int): Destination rank
+        op (optional): One of the values from
+            ``torch.distributed.ReduceOp``
+            enum.  Specifies an operation used for element-wise reductions.
+        group (ProcessGroup, optional): The process group to work on
+        dst_tensor (int, optional): Destination tensor rank within
+                                    ``tensor_list``
+
+    Returns:
+        list[Tensor]: received tensors from the collective
+
+    """
+    return _ReduceMultiGPU.apply(dst, op, group, dst_tensor, *tensors)
+
+
+def all_gather(tensor, group=dist.group.WORLD):
+    """
+    Gathers tensors from the whole group in a list.
+
+    Arguments:
+        tensor (Tensor): Tensor to be broadcast from current process.
+        group (ProcessGroup, optional): The process group to work on
+
+    Returns:
+        list[Tensor]): output of the collective
+
+    """    
+    return _AllGather.apply(group, tensor)
+
+
+def all_to_all(tensors, group=dist.group.WORLD):
+    """
+    Each process scatters list of input tensors to all processes in a group and
+    return gathered list of tensors in output list.
+
+    Arguments:
+        tensors (list[Tensor]): List of tensors to scatter one per rank.
+        group (ProcessGroup, optional): The process group to work on.
+
+    Returns:
+        list[Tensor]): output of the collective
+
+    """
+    return _AlltoAll.apply(group, *tensors)
+
+
+def all_reduce(tensor, op=dist.ReduceOp.SUM, group=dist.group.WORLD):
+    """
+    Reduces the tensor data across all machines in such a way that all get
+    the final result.
+
+    After the call ``tensor`` is going to be bitwise identical in all processes.
+
+    Arguments:
+        tensor (Tensor): Input of the collective.
+        op (optional): One of the values from
+            ``torch.distributed.ReduceOp``
+            enum.  Specifies an operation used for element-wise reductions.
+        group (ProcessGroup, optional): The process group to work on
+
+    Returns:
+        Tensor: output of the collective
+
+    """
+    return _AllReduce.apply(op, group, tensor)
+
+
+class _Broadcast(Function):
+    @staticmethod
+    def forward(ctx, src, group, tensor):
+        ctx.src = src
+        ctx.group = group
+        ctx.rank = dist.get_rank()
+        dist.broadcast(tensor, src, group=group)
+        return tensor
+
+    @staticmethod
+    def backward(ctx, *grad_outputs):
```

Comment:
There is a single `grad_outputs` here right?
You culd remove the `*` here and on the next line.